src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2016,22 +2016,27 @@ impl Coordinator {
             // data. We "cheat" a little bit and filter out any IDs that aren't
             // user objects because we know they are not a RTR source.
             let mut to_visit = VecDeque::from_iter(source_ids.filter(GlobalId::is_user));
-            let mut visited = BTreeSet::new();
+            // If none of the sources are user objects, we don't support RTR.
```

Comment:
```suggestion
            // If none of the sources are user objects, we don't need to provide a RTR timestamp.
```
What happens in the caller though? The query returns immediately? Or times out because it can't get a timestamp to wait for?

Reply:
Do we have a test case for this? If not, could one be added easily?

Reply:
> What happens in the caller though? The query returns immediately? Or times out because it can't get a timestamp to wait for?

Their query is not influenced by any external system's frontier, so the RTR machinery just isn't involved in determining a timestamp. We can test this by querying something that doesn't depend on a user source and show that it returns values––certainly doesn't hurt to demonstrate.